src/app/util/af.h

```diff
@@ -101,10 +101,23 @@ bool emberAfContainsClient(chip::EndpointId endpoint, chip::ClusterId clusterId)
  * over the air. Because this function is being called locally
  * it assumes that the device knows what it is doing and has permission
  * to perform the given operation.
+ *
+ * This function also does NOT check that the input dataType matches the expected
+ * data type (as Accessors.h/cpp have this correct by default).
+ * TODO: this not checking seems off - what if this is run without Accessors.h ?
  */
 EmberAfStatus emberAfWriteAttribute(chip::EndpointId endpoint, chip::ClusterId cluster, chip::AttributeId attributeID,
                                     uint8_t * dataPtr, EmberAfAttributeType dataType);
 
+/**
+ * Write an attribute for a request arriving from external sources.
+ *
+ * This will check attribute attribute writeability and that
+ * the privided data type matches the expected data type.
+ */
+EmberAfStatus emberAfWriteAttributeExternal(chip::EndpointId endpoint, chip::ClusterId cluster, chip::AttributeId attributeID,
```

Comment:
This isn't really public API, right?  As in, applications should not be calling this, ever.  Why did this declaration get moved here?

Reply:
because it starts with `ember` in the previous version, so I moved it here. I am not familiar enough with ember right now, so my instinct was: `emAf` are internal, `emberAf` are external. Are there any conventions in this codebase that I can follow or at least start enforcing?

To me the layout is generally not ideal: having a separate header that exposes things define in different cpp files is not ideal. We should have had `af.cpp`.

Reply:
Moved back to old location, but renamed as `emAfWriteAttributeExternal` to try to have more of `emberAf` is public and `emAf` is internal.... 

Overall I believe apps should generally never call `em*` at all and should instead use Accessors.h and similar - that is the route that I am trying to move towards.